api/identityfile/identityfile.go

```diff
@@ -261,7 +262,7 @@ func decodeIdentityFile(idFile io.Reader) (*IdentityFile, error) {
 	// are copied out of the scanner's buffer.  All others are ignored.
 	for scanln() {
 		switch {
-		case hasPrefix("ssh"):
+		case isSSHCert(line):
```

Comment:
Thinking about this again, would it be too slow to `ssh.ParseAuthorizedKeys` and check that the public key we got is a `*ssh.Certificate`?

Reply:
I think the performance difference would be negligible. Honestly, `decodeIdentityFile` should probably just fully parse the entire file into it's actual `ssh.Certificate`, `tls.Certificate`, `keys.PrivateKey`, etc, since they will need to be parsed (probably multiple times) afterwards.

Reply:
On second thought, I think this is out of scope, since currently the decode function is is just meant to parse and organize the identity file parts, rather than verify the contents. For example, if there is an ssh cert present, but it's invalid, decode should not return "ssh cert missing". Instead, we should parse it after decoding.